turbo/trie/hashbuilder.go

```diff
@@ -531,6 +579,12 @@ func (hb *HashBuilder) branchHash(set uint16) error {
 	if _, err := hb.sha.Read(hb.hashStack[len(hb.hashStack)-length2.Hash:]); err != nil {
 		return err
 	}
+
+	if doProof && (hb.accProofResult != nil) {
+		nodeBytes := hexutil.Bytes(proofBuf.Bytes())
+		hb.accProofResult.AccountProof = append([]string{nodeBytes.String()}, hb.accProofResult.AccountProof...)
```

Comment:
- similar question: possible to call `e.TopHashes()` or `e.TopNodes()`(top of e.nodeStack) - instead of modifying hash builder?

Reply:
See my latest comment on the previous issue; I've reduced the scope of the required hashbuilder changes but I don't think that e.nodeStack would help to make it any simpler than what I have now.